File: 151002/Saprin/RV/src/main/java/by/bsuir/rv/service/editor/IEditorService.java
package by.bsuir.rv.service.editor;

import by.bsuir.rv.dto.EditorRequestTo;
import by.bsuir.rv.dto.EditorResponseTo;
import by.bsuir.rv.exception.EntityNotFoundException;

import java.math.BigInteger;
import java.util.List;

public interface IEditorService {

    List<EditorResponseTo> getEditors();
    EditorResponseTo getEditorById(BigInteger id) throws EntityNotFoundException;
    EditorResponseTo createEditor(EditorRequestTo editor);
    EditorResponseTo updateEditor(EditorRequestTo editor) throws EntityNotFoundException;
    void deleteEditor(BigInteger id) throws EntityNotFoundException;

}

File: 151003/Mitrofanov/Publisher/Service/Interface/IEditorService.cs
using Publisher.Entity.Db;
using Publisher.Entity.DTO.RequestTO;
using Publisher.Entity.DTO.ResponseTO;
using Publisher.Service.Interface.Common;

namespace Publisher.Service.Interface
{
    public interface IEditorService : ICrudService<Editor, EditorRequestTO, EditorResponseTO>
    {
        Task<EditorResponseTO> GetByStoryID(int storyId);
    }
}
